airbyte-scheduler/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -287,9 +287,14 @@ private Job getJob(DSLContext ctx, long jobId) {
   }
 
   @Override
-  public Optional<Job> getOldestPendingJob() throws IOException {
+  public Optional<Job> getNextJob() throws IOException {
+    // rules:
```

Comment:
a little unclear if 2 and 3 should apply for jobs that are not sync jobs. e.g. there's no reason why two spec queries need to wait on each other. i'm inclined to say it doesn't matter, but if we do run into a performance issue here we can changes it so rules 2 and 3 _only_ apply to sync jobs. inclined to keep it simpler until we need to though.

Reply:
I agree with you that we might want to always resolve jobs that are not syncs. 

The question is: in what scenario can we have two sync jobs for the same config in a non terminal state? That doesn't seem right.

Reply:
Actually read through the issue. 

Why did you chose this option instead of having a conditional add when someone asks for a manual sync? and not do it if there is a sync running?

Reply:
It is definitely important to enforce this inside the job submitter so we don't have undefined behavior. I want this invariant to be bullet proof at the core level.

Currently the only case this comes up is if the user hits the sync now button a bunch of times. I have no problem adding logic there as well that if something is already running, now job is added if we want to prevent queuing multiple jobs. I think there's some complexity there. If a user clicks "sync now" while a reset is running do we want to prevent them from queueing the job? It is not obvious to me that we do. 

Either the behavior is that every time you hit the "sync now" or "reset job" button we queue a job or when you hit those buttons we queue a job if it is allowed and provide an error message. no strong preference.